packages/flutter/lib/src/painting/debug.dart

```diff
@@ -31,6 +32,99 @@ typedef HttpClientProvider = HttpClient Function();
 /// This value is ignored in non-debug builds.
 HttpClientProvider debugNetworkImageHttpClientProvider;
 
+typedef PaintImageCallback = void Function(ImageSizeInfo);
+
+/// Tracks the bytes used by a [ui.Image] compared to the bytes needed to paint
+/// that image without scaling it.
+@immutable
+class ImageSizeInfo {
+  /// Creates an object to track the backing size of a [ui.Image] compared to
+  /// its display size on a [Canvas].
+  ///
+  /// This class is used by the framework when it paints an image to a canvas
+  /// to report to `dart:developer`'s [postEvent], as well as to the
+  /// [debugOnPaintImage] callback if it is set.
+  const ImageSizeInfo({this.source, this.displaySize, this.imageSize});
+
+  /// A unique identifier for this image, for example its asset path or network
+  /// URL.
+  final String source;
+
+  /// The size of the area the image will be rendered in.
+  final Size displaySize;
+
+  /// The size the image has been decoded to.
+  final Size imageSize;
+
+  /// The number of bytes needed to render the image without scaling it.
+  int get displaySizeInBytes => _sizeToBytes(displaySize);
+
+  /// The number of bytes used by the image in memory.
+  int get decodedSizeInBytes => _sizeToBytes(imageSize);
+
+  int _sizeToBytes(Size size) {
+    return (size.width * size.height * 4 * (4/3)).toInt();
```

Comment:
This is assuming 16/3 bytes per pixel which is surprising unless I'm missing something. 
Can you document why it is `4 * (4/3)`?
I would have expected either 4 bytes per pixel or 3 bytes per pixel depending on if you are assuming an alpha channel.

Reply:
Added a comment. The 4/3 is mipmapping overhead, which we turn on if available in the engine - and really it's almost always on, unless you're doing software rendering.